src/Symfony/Bridge/Doctrine/Logger/DbalLogger.php

```diff
@@ -49,12 +49,8 @@ public function startQuery($sql, array $params = null, array $types = null)
             $this->stopwatch->start('doctrine', 'doctrine');
         }
 
-        if (is_array($params)) {
-            $params = $this->normalizeParams($params);
-        }
-
         if (null !== $this->logger) {
-            $this->log($sql, null === $params ? array() : $params);
+            $this->log($sql, null === $params ? array() : $this->normalizeParams($params));
```

Comment:
If `$params` is not an array and not `null`; PHP will raise a fatal error because the function signature states `array`.

sth like;
```
PHP Catchable fatal error:  Argument 2 passed to startQuery() must be of the type array, integer given, ...
```

so the test for `array` and later on for `null` was overcomplete AFAIK